### PR TITLE
Improved help and added a test for it.

### PR DESCRIPTION
**Pull request by SamMousa**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #10372

### CODE REVIEW

**Comment by SamMousa** [comment]
Hmm, thanks for fixing that will take it into account in the future :)

File: tests/framework/console/ControllerTest.php [hunk 1]
```diff
@@ -132,4 +132,18 @@ public function testHelpOptionWithModule()
         $this->assertFalse(FakeController::getWasActionIndexCalled());
         $this->assertEquals(FakeHelpController::getActionIndexLastCallParams(), ['news/posts/index']);
     }
+
+
+    /**
+     * Tests if action help does not include (class) type hinted arguments.
+     * @see #10372
+     */
+    public function testHelp()
```
Comment (samdark) [inline tests/framework/console/ControllerTest.php]: Could be named `testHelpSkipsTypehintedArguments()`. Long but name itself explains what it does.

**Review by samdark — CHANGES_REQUESTED** [review]

**Review by samdark — APPROVED** [review]

File: tests/framework/console/ControllerTest.php [hunk 2]
```diff
@@ -132,4 +132,18 @@ public function testHelpOptionWithModule()
         $this->assertFalse(FakeController::getWasActionIndexCalled());
         $this->assertEquals(FakeHelpController::getActionIndexLastCallParams(), ['news/posts/index']);
     }
+
+
+    /**
+     * Tests if action help does not include (class) type hinted arguments.
+     * @see #10372
+     */
+    public function testHelpSkipsTypehintedArguments()
```
Comment (arogachev) [inline tests/framework/console/ControllerTest.php]: Maybe ...Type**H**inted... ?

File: tests/framework/console/FakeController.php [hunk 3]
```diff
@@ -84,6 +84,10 @@ public function actionAksi6()
         return $this->testArray;
     }
 
+    public function actionAksi7(self $typedArgument, $simpleArgument) {
```
Comment (arogachev) [inline tests/framework/console/FakeController.php]: Sorry, but what `aksi7` means?

**Review by arogachev — CHANGES_REQUESTED** [review]

File: tests/framework/console/FakeController.php (hunk 3 above)
Comment (SamMousa) [inline tests/framework/console/FakeController.php]: It's an action in the fake controller.

Comment (dynasource) [inline tests/framework/console/FakeController.php]: can you rename this action towards a name that is less magical and closer related to the actual test scenario?

**Review by dynasource — CHANGES_REQUESTED** [review]

**Comment by SamMousa** [comment]
You mean like aksi1 through aksi6? *-)

**Comment by dynasource** [comment]
`actionWithDifferentArgumentHints'

**Comment by SamMousa** [comment]
I understand what you mean, I'm just saying that the other 6 action follow
a different convention that I chose to follow. 8-)

File: tests/framework/console/FakeController.php (hunk 3 above)
Comment (SamMousa) [inline tests/framework/console/FakeController.php]: Please have a look at the rest of `FakeController`. It contains action that are numerically named for some reason. If you don't like it you should fix the whole controller not just 1 random function.

**Comment by SamMousa** [comment]
This is ready for merge.

**Comment by samdark** [comment]
@dynasource fixing naming in existing non-related tests is out of scope of this pull request.

File: tests/framework/console/FakeController.php (hunk 3 above)
Comment (samdark) [inline tests/framework/console/FakeController.php]: The controller was like that and renaming requires much more analysis. I'd say leave it as is for now.

Comment (dynasource) [inline tests/framework/console/FakeController.php]: for my approval, this newly added testAction should be altered based on my argument https://github.com/yiisoft/yii2/pull/13772/files#r106026067

**Review by dynasource — CHANGES_REQUESTED** [review]

File: tests/framework/console/FakeController.php (hunk 3 above)
Comment (samdark) [inline tests/framework/console/FakeController.php]: Right, it's a new one. Then it makes sense to name it properly.

**Review by samdark — CHANGES_REQUESTED** [review]

**Review by dynasource — APPROVED** [review]

File: tests/framework/console/FakeController.php (hunk 3 above)
Comment (samdark) [inline tests/framework/console/FakeController.php]: 1 function is a good start. We'll fix the rest sooner or later.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
https://travis-ci.org/yiisoft/yii2/jobs/211268219#L311

**Comment by SamMousa** [comment]
GitHub IDE is not ideal. Works again now.

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**